packages/docusaurus-types/src/index.d.ts

```diff
@@ -21,6 +21,12 @@ export type ThemeConfig = {
   [key: string]: unknown;
 };
 
+export interface UserDocusaurusConfig extends Partial<DocusaurusConfig> {
```

Comment:
I'd rather not redeclare types here.

We already use this utility: https://github.com/piotrwitek/utility-types#requiredt-k

This should work: `Required<Partial<DocusaurusConfig>,"baseUrl" | "url" | "title">`

Reply:
That's elegant.